docs/en/observability/profiling-self-managed-ops.asciidoc

```diff
@@ -185,44 +186,80 @@ When scaling out the number of replicas, depending on the load balancer that you
 This ensures that the load is evenly distributed across all replicas.
 
 [discrete]
-[[profiling-upgrade-backend-bin]]
-== Upgrade the backend binaries
+[[profiling-self-managed-upgrade]]
+== Upgrades
 
-Upgrade the backend binaries whenever you upgrade the rest of the Elastic stack.
-While we try to keep backward compatibility between two consecutive minor version, we may introduce changes to the data format that require the applications to be at the same version of Elasticsearch and Kibana.
+Upgrading a self-hosted stack involves upgrading the backend applications and the agent.
+We recommend to upgrade the backend first, followed by the agent. In this scenario, if the backend has problems,
+rolling back to the previous version is the safest option and does not require the agent to be downgraded.
+
+NOTE: It is usually best to have the same version of the agent and the backend deployed together.
+
+We strive to maintain backward compatibility between minor versions.
+We may introduce changes to the data format that require the agent and backend to be at the same version.
```

Comment:
```suggestion
Some changes to the data format may require having the same version of the agent and backend deployed.
```

Reply:
I would emphasize this is rarely happening, can we add "Occasionally"?

```suggestion
Occasionally, changes to the data format may require having the same version of the agent and backend deployed.
```

Reply:
Works for me.